doc/decisions/constructor_functions_examples.md

```diff
@@ -0,0 +1,105 @@
+### Constructor Functions: Language examples
```

Comment:
Please note: This file is just for illustration, I will probably remove it before we merge the decision.